tests/serialization/test_states.py

```diff
@@ -338,22 +298,70 @@ def test_new_result_with_location_deserializes_correctly(self):
         assert isinstance(new_state._result, results.S3Result)
         assert new_state._result.location == "dir/place.txt"
 
-    def test_cached_inputs_are_serialized_correctly(self):
-        s = state.Cached(
-            message="test",
-            result=results.PrefectResult(value=1, location="1"),
-            cached_inputs=dict(
-                x=results.PrefectResult(location='"foo"'),
-                y=results.PrefectResult(location='"bar"'),
-            ),
-        )
-        schema = StateSchema()
-        serialized = schema.dump(s)
-
-        assert serialized["cached_inputs"]["x"]["location"] == '"foo"'
-        assert serialized["cached_inputs"]["y"]["location"] == '"bar"'
 
-        new_state = schema.load(serialized)
+@pytest.mark.parametrize(
+    "old_json",
+    [
+        {
+            "type": "Success",
+            "_result": {"type": "NoResultType", "__version__": "0.6.0"},
+            "message": "Task run succeeded.",
+            "__version__": "0.6.0",
+        },
+        {
+            "type": "Mapped",
+            "context": {"tags": []},
+            "message": "Mapped tasks submitted for execution.",
+            "__version__": "0.7.1+108.g3ad8ced1",
+            "n_map_states": 2,
+        },
+        {
+            "type": "Success",
+            "_result": {
+                "type": "SafeResult",
+                "value": '["test1.zip", "test2.zip"]',
+                "__version__": "0.7.1+108.g3ad8ced1",
+                "result_handler": {
+                    "type": "JSONResultHandler",
+                    "__version__": "0.7.1+108.g3ad8ced1",
+                },
+            },
+            "context": {"tags": []},
+            "message": "Task run succeeded.",
+            "__version__": "0.7.1+108.g3ad8ced1",
+        },
+        {
+            "type": "Success",
+            "_result": {
+                "type": "SafeResult",
+                "value": "/src/prefect-result-2020-06-09t15-32-47-297967-00-00",
+                "__version__": "0.11.5+59.gdba764390",
+                "result_handler": {
+                    "dir": "/src",
+                    "type": "LocalResultHandler",
+                    "__version__": "0.11.5+59.gdba764390",
+                },
+            },
+            "context": {"tags": []},
+            "message": "Task run succeeded.",
+            "__version__": "0.11.5+59.gdba764390",
+            "cached_inputs": {
+                "report": {
+                    "type": "SafeResult",
+                    "value": "/src/prefect-result-2020-06-09t15-32-38-403718-00-00",
+                    "__version__": "0.11.5+59.gdba764390",
+                    "result_handler": {
+                        "dir": "/src",
+                        "type": "LocalResultHandler",
+                        "__version__": "0.11.5+59.gdba764390",
+                    },
+                }
+            },
+        },
+    ],
+)
+def test_can_deserialize_old_no_result(old_json):
```

Comment:
This test is a new test that actually confirms we can deserialize old state payloads still